### PR TITLE
MNT Add non-strict mypy to CI and fix errors

### PR DESCRIPTION
**Pull request by BenjaminBossan**
This accompanies PR #61 to contrast the difference between strict and
non-strict mypy usage.

### CODE REVIEW

File: skops/hub_utils/_hf_hub.py [hunk 1]
```diff
@@ -112,7 +114,7 @@ def init(*, model: Union[str, Path], requirements: List[str], dst: Union[str, Pa
     None
     """
     dst = Path(dst)
-    if dst.exists() and next(dst.iterdir(), None):
+    if dst.exists() and bool(next(dst.iterdir(), None)):
```
Comment (adrinjalali) [inline skops/hub_utils/_hf_hub.py:117]: aha? why do we need explicit bool here? This is proper python:

``` python
>>> if 3 or None:
...     print(1)
```

**Review by adrinjalali — COMMENTED** [review]
This looks much more reasonable to me.

Should also add it to pre-commit hooks.

File: skops/hub_utils/_hf_hub.py (hunk 1 above)
Comment (BenjaminBossan) [reply skops/hub_utils/_hf_hub.py:117]: The error is:

> skops/hub_utils/_hf_hub.py:117: error: Argument 1 to "next" has incompatible type "Generator[Path, None, None]"; expected "SupportsNext[bool]"
> skops/hub_utils/_hf_hub.py:117: error: Argument 2 to "next" has incompatible type "None"; expected "bool"

Comment (adrinjalali) [reply skops/hub_utils/_hf_hub.py:117]: That's very silly.

**Review by adrinjalali — COMMENTED** [review]
If you add the pre-commit hook here, I'm happy to merge this as a first step, and re-visit in the future.

File: skops/_min_dependencies.py [hunk 2]
```diff
@@ -16,7 +16,8 @@
     "pytest": (PYTEST_MIN_VERSION, "tests", None),
     "pytest-cov": ("2.9.0", "tests", None),
     "flake8": ("3.8.2", "tests", None),
-    "mypy": ("0.770", "tests", None),
+    "mypy": ("0.971", "tests", None),
```
Comment (adrinjalali) [inline skops/_min_dependencies.py:19]: mypy isn't really a test dependency. It probably makes more sense to have it explicitly installed in CI the same as black and isort.

**Review by adrinjalali — APPROVED** [review]
Let's merge and see how it goes.

**Pull request merged**